lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -72,6 +72,31 @@ static bool isDuplicatableNullaryExpression(Operation *op) {
   return false;
 }
 
+/// Return the verilog name of the operations that can define a symbol.
+static StringRef getSymOpName(Operation *symOp) {
+  // Typeswitch of operation types which can define a symbol.
+  return TypeSwitch<Operation *, StringRef>(symOp)
+      .Case<HWModuleOp>([&](HWModuleOp op) { return op.getName(); })
+      .Case<HWModuleExternOp>(
+          [&](HWModuleExternOp op) { return op.getVerilogModuleName(); })
+      .Case<HWGeneratorSchemaOp>(
+          [&](HWGeneratorSchemaOp op) { return op.sym_name(); })
+      .Case<InstanceOp>([&](InstanceOp op) { return op.getName().getValue(); })
+      .Case<WireOp>([&](WireOp op) { return op.name(); })
+      .Case<RegOp>([&](RegOp op) { return op.name(); })
+      .Case<InterfaceOp>([&](InterfaceOp op) {
+        return getVerilogModuleNameAttr(op).getValue();
+      })
+      .Case<InterfaceSignalOp>(
+          [&](InterfaceSignalOp op) { return op.sym_name(); })
+      .Case<InterfaceModportOp>(
+          [&](InterfaceModportOp op) { return op.sym_name(); })
+      .Default([&](Operation *op) {
+        op->emitError("unknown operation with symbol, cannot determine name");
```

Comment:
Actually, this should emit the error on the sv.verbatim* operation, and add a note that points to "op".  Please pass the location (or the op) of the verbatim into this so it can be diagnosed properly.   Please add a testcases for this.  You can do this by dropping an std.func into the top level that defines a symbol and referring to it.

Reply:
Moved the error to the caller instead, to error on the verbatim op.